drivers/misc/devmux/Kconfig

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2023, Meta
+# SPDX-License-Identifier: Apache-2.0
+
+config DEVMUX
+	bool "Device Multiplexer (devmux) [EXPERIMENTAL]"
+	depends on DT_HAS_ZEPHYR_DEVMUX_ENABLED
+	depends on DEVICE_MUTABLE
+	select EXPERIMENTAL
+	help
+	  Devmux is a pseudo-device that operates as a device switch. It allows
+	  software to select the data, config, and api from a number of linked
+	  devices.
+
+if DEVMUX
```

Comment:
Nit: make it a `depends on` instead.

Reply:
I usually see `depends on` when the dependency is from a separate `Kconfig` file. It's more common to use `if FOO` in the same file where `FOO` is defined, because it saves space when there are (or it is anticipated that there will be) multiple dependencies in the same file.